### PR TITLE
Lesson9 solution

### PR DESCRIPTION
**Pull request by Ognessa**

### CODE REVIEW

File: app/src/main/java/com/onix/internship/objects/DataStore.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    fun editNote(note : NotesData){
+        db.notesDao().update(note)
+    }
+
+    fun clearDB(){
+        db.notesDao().apply {
+            deleteAll()
+            deleteKey()
```
Comment (shadwork) [inline app/src/main/java/com/onix/internship/objects/DataStore.kt:32]: no needed in key deletion - one table - so - no relation between tables

File: app/src/main/java/com/onix/internship/objects/NotesColors.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+fun getColorsList(): Array<out NotesColors>? {
+    return NotesColors::class.java.enumConstants
+}
+
+fun getColor(color: NotesColors):  Int {
+    return when (color) {
+        NotesColors.RED -> Color.RED
+        NotesColors.YELLOW -> Color.YELLOW
+        NotesColors.GREEN -> Color.GREEN
+        NotesColors.PINK -> Color.argb(255, 254, 79, 172)
```
Comment (shadwork) [inline app/src/main/java/com/onix/internship/objects/NotesColors.kt:18]: colors should be in xml values

File: app/src/main/java/com/onix/internship/ui/addNote/AddNoteFragment.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        viewModel.saveNoteEvent.observe(this){
+            viewModel.saveNote(args.fragmentState)
+        }
+        viewModel.returnToMenu.observe(this){
+            navigateToPrevious()
+        }
+    }
+
+    private fun createColorMenu(llColorsListContainer: LinearLayout) {
```
Comment (shadwork) [inline app/src/main/java/com/onix/internship/ui/addNote/AddNoteFragment.kt:70]: good but rewrite it with recycleview

File: app/src/main/java/com/onix/internship/ui/addNote/AddNoteViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import com.onix.internship.objects.NotesData
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class UpdateTextModel(val title: String, val description: String)
+
+class AddNoteViewModel(private val dataStore: DataStore, private val application: Application) :
+    BaseViewModel() {
+
+    private val _note = MutableLiveData<NotesData>()
```
Comment (shadwork) [inline app/src/main/java/com/onix/internship/ui/addNote/AddNoteViewModel.kt:21]: thinking about incapsulate all this staff into model with observable fields

**Review by shadwork — CHANGES_REQUESTED** [review]
неплохо - но мало

File: app/src/main/java/com/onix/internship/objects/DataStore.kt (hunk 1 above)
Comment (Ognessa) [reply app/src/main/java/com/onix/internship/objects/DataStore.kt]: Done

File: app/src/main/java/com/onix/internship/objects/NotesColors.kt (hunk 2 above)
Comment (Ognessa) [reply app/src/main/java/com/onix/internship/objects/NotesColors.kt]: Done

**Review by shadwork — APPROVED** [review]

**Pull request merged**